tutorials/haskell/haskell-time-tutorial/tutorial.md

```diff
@@ -0,0 +1,535 @@
+```
+
+As you can see, it is only an alias for `Int`. We now are able to define
+months like `Int`s where January will correspond to 1 and so on. But
+fortunately, since GHC 7.8 we are able to use pattern synonyms and we
+could, instead of using raw `Int`s, write this down in a pattern style
+and have a more idiomatic way of defining months:
+
+```haskell
+pattern January :: MonthOfYear
+pattern January = 1
+
+pattern February :: MonthOfYear
+pattern February = 2
+
+pattern March :: MonthOfYear
+pattern March = 3
+.
+.
+.
+
+-- | The twelve 'MonthOfYear' patterns form a @COMPLETE@ set.
+pattern December :: MonthOf
+pattern December = 12
+
+{-# COMPLETE January, February, March, April, May, June, July, August, September, October, November, December #-}Year
+```
+
+If you’re curious about why the `{-# COMPLETE #-}` pragma is being
+used, see the [COMPLETE pragma documentation][complete-pragma] for more
+information.
+
+Just to make sure that `November` and `11` are equivalent, let's test it
+in a stack repl:
+
+```bash
+> November == 11
+True
+```
+
+We’ll use this pattern in a function that receives a `Day` and yields if the
+corresponding Date is a holiday:
+
+```haskell
+dayToHoliday :: Day -> String
+dayToHoliday (YearMonthDay _ January  _)  = "Happy new year!"
+dayToHoliday (YearMonthDay _ November 1)  = "Let's eat colada morada"
+dayToHoliday (YearMonthDay _ December 25) = "Merry Christmas jojojo"
+dayToHoliday (YearMonthDay _ _ _)         = "Probably just a regular day"
+```
+
+Another pattern synonym is introduced here. The `YearMonthDay` pattern lets us
+build a `Day` in terms of the `Year`, `MonthOfYear` (do you remember this one?)
+and `DayOfMOnth`, where all three are basically `Int`s. Here is its definition:
+
+```haskell
+pattern YearMonthDay :: Year -> MonthOfYear -> DayOfMonth -> Day
+pattern YearMonthDay y m d <-
+   (toGregorian -> (y, m, d))
+   where
+       YearMonthDay y m d = fromGregorian y m d
+```
+
+Back to our function, let's check that is working:
+
+```haskell
+> dayToHoliday (YearMonthDay 2021 11 01)
+"Let's eat colada morada"
+
+> dayToHoliday (YearMonthDay 2021 November 01)
+"Let's eat colada morada"
+```
+
+(Now you're probably wondering what is “colada morada”? Haha no worries,
+it’s an [Ecuatorian tradition][colada-morada] and we eat this meal that
+is purple corn based with spices and fruits, pretty amazing to be honest.)
+
+Back to Haskell, I think it’s pretty clear what the `dateToHoliday`
+function does, the first clause matches all days of January, second
+one matches the All Saints holiday, the third one matches for Christmas
+and the last one it’s the fallback for all other days. I think using
+the `MonthOfYears` patterns makes it a lot more readable. Having something
+like:
+
+```haskell
+dateToHoliday 12 25  = "Merry Christmas jojojo"
+```
+
+Will not be as straightforward as the function that we have right now.
+
+I haven’t mentioned it yet, but the patterns that we built previously
```

Comment:
Let's try to write all the sentences in the third-person point of view.

Reply:
@sestrella Can you elaborate on this comment? I'm not sure if you mean something about the “I” or the “we” here. Just making sure this is clear.